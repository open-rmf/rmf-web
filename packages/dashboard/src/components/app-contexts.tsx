import React from 'react';
import ResourceManager from '../managers/resource-manager';
import { defaultSettings, Settings } from '../settings';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceManager | undefined>(undefined);

export const SettingsContext = React.createContext(defaultSettings());

export interface AppController {
  setSettings: React.Dispatch<React.SetStateAction<Settings>>;
  saveSettings(settings: Settings): void;
  showSettings(show: boolean): void;
  toggleSettings(): void;
  showHelp(show: boolean): void;
  toggleHelp(): void;
  showHotkeysDialog(show: boolean): void;
  toggleHotkeysDialog(): void;
  /**
   * FIXME: Move this to settings, this sets if we should show the tooltip icon, and not trigger
   * a tooltip to show.
   */
  showTooltips(show: boolean): void;
  /**
   * FIXME: Move this to settings, this sets if we should show the tooltip icon, and not trigger
   * a tooltip to show.
   */
  toggleTooltips(): void;
}

export interface Tooltips {
  showTooltips: boolean;
}

export const TooltipsContext = React.createContext<Tooltips>({
  showTooltips: true,
});

export const AppControllerContext = React.createContext<AppController>({
  setSettings: () => {},
  saveSettings: () => {},
  showSettings: () => {},
  toggleSettings: () => {},
  showHelp: () => {},
  toggleHelp: () => {},
  showHotkeysDialog: () => {},
  toggleHotkeysDialog: () => {},
  showTooltips: () => {},
  toggleTooltips: () => {},
});

export interface AppContent {
  tabNames: string[];
}
export const AppContentContext = React.createContext<AppContent>({ tabNames: ['Building'] });
