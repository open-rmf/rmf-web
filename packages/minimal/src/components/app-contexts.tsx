import React from 'react';
import appConfig, { AppConfig } from '../app-config';
import dataConfig, { DataConfig } from '../config/data-config';
import { ResourceManager } from 'rmf-tools';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceManager | undefined>(undefined);

export const AppConfigContext = React.createContext<AppConfig>(appConfig);

export const DataConfigContext = React.createContext<DataConfig>(dataConfig);
