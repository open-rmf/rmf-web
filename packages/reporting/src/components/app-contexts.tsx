import React from 'react';
import appConfig, { AppConfig } from '../app-config';

export const AppConfigContext = React.createContext<AppConfig>(appConfig);
