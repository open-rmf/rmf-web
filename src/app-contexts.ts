import { createContext } from 'react';
import { ResourceConfigurationsType } from './resource-manager';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = createContext<ResourceConfigurationsType>({});
