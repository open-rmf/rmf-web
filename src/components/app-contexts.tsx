import React from 'react';
import { ResourceConfigurationsType } from '../resource-manager';
import { defaultSettings, Settings } from '../settings';
import { NotificationBarContext, NotificationBarProps } from './notification-bar';

/* Declares the ResourcesContext which contains the resources used on the app*/
export const ResourcesContext = React.createContext<ResourceConfigurationsType>({});

export const SettingsContext = React.createContext(defaultSettings());

export interface AppContextProviderProps extends React.PropsWithChildren<{}> {
  settings: Settings;
  notificationDispatch: React.Dispatch<React.SetStateAction<NotificationBarProps | null>>;
}

export function AppContextProvider(props: AppContextProviderProps): React.ReactElement {
  const { settings, notificationDispatch, children } = props;
  return (
    <SettingsContext.Provider value={settings}>
      <NotificationBarContext.Provider value={notificationDispatch}>
        {children}
      </NotificationBarContext.Provider>
    </SettingsContext.Provider>
  );
}
