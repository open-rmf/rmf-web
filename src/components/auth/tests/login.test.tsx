import { createMount } from '@material-ui/core/test-utils';
import React from 'react';
import { Redirect } from 'react-router';
import { BrowserRouter } from 'react-router-dom';
import { UserContext } from '../../../app-contexts';
import Login from '../login';

const mount = createMount();

describe('Login page', () => {
  test('renders correctly', () => {
    const component = mount(
      <BrowserRouter>
        <Login />
      </BrowserRouter>,
    );
    expect(component.html()).toMatchSnapshot();
  });

  test('redirects to dashboard when user is authenticated', async () => {
    window.history.replaceState(window.history.state, '');
    const wrapper = mount(
      <BrowserRouter>
        <UserContext.Provider
          value={{
            username: 'test',
          }}
        >
          <Login />
        </UserContext.Provider>
      </BrowserRouter>,
    );
    expect(wrapper.find(Redirect)).toBeTruthy();
  });
});
