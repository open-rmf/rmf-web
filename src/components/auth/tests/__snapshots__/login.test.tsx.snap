// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly 1`] = `"<div class=\\"makeStyles-flexColumnContainer-1 makeStyles-fullPage-2\\"><div class=\\"makeStyles-flexColumnContainer-1 makeStyles-authContainer-3\\"><h4 class=\\"MuiTypography-root makeStyles-authTitle-4 MuiTypography-h4\\">RoMi Dashboard</h4><img src=\\"assets/ros-health.png\\" alt=\\"\\" class=\\"makeStyles-logo-8\\"><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-contained\\" tabindex=\\"0\\" type=\\"button\\" id=\\"login-button\\"><span class=\\"MuiButton-label\\">Login with RMF</span><span class=\\"MuiTouchRipple-root\\"></span></button></div></div>"`;
